WordPress/openverse-api
# Add Django database migrations to Dockerfile entrypoint

## Fixes




Fixes #1048 by @AetherUnbound

## Description




This PR adds the Django database migration to the Dockerfile entrypoint so it occurs automatically on startup. This can be disabled by setting `DJANGO_MIGRATE_DB_ON_STARTUP=False`.

> **Note**
> We'll want to add the disable flag to the current EC2-based deployment setup, since all of the instances in the auto-scaling group start at the same time which we want to avoid (see https://github.com/WordPress/openverse-infrastructure/issues/176).

## Testing Instructions



1. Run `just build && just up && just logs web` and check that the "Running database migrations..." line does not show up in the logs.
1. Comment out the `DJANGO_MIGRATE_DB_ON_STARTUP` line in `api/env.docker` then run `just up && just logs web` and check that the "Running database migrations..." line _does_ show up in the logs.
1. (Optional) Run `just dj migrate api 0050` to roll back a migration. Then run `just down && just up && just logs web` and check that the migration is run automatically.

## Checklist



      `Update index.md`).
      a parent feature branch.
      errors.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin



<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
